modules/tinymce/src/themes/silver/main/ts/ui/dialog/Tree.ts

```diff
@@ -75,84 +166,130 @@ const renderDirectoryLabel = (directory: Dialog.Directory, visible: boolean, bac
+      classes: [ 'tox-tree--directory__label', 'tox-trbtn' ].concat( visible ? [ 'tox-tree--directory__label--visible' ] : [] ),
     },
     components,
-    action: (button) => {
-      SelectorFind.sibling(button.element, '.tox-tree--directory__children').each((childrenEle) => {
-        button.getSystem().getByDom(childrenEle).each((childrenComp) => Toggling.toggle(childrenComp));
-      });
-      SelectorFind.ancestor(button.element, '.tox-tree--directory').each((directoryEle) => {
-        button.getSystem().getByDom(directoryEle).each((directoryComp) => Toggling.toggle(directoryComp));
-      });
+    action: expandChildren,
+    eventOrder: {
+      [NativeEvents.keydown()]: [
+        directoryLabelEventsId,
+        'keying',
+      ]
     },
     buttonBehaviours: Behaviour.derive([
-      Toggling.config({
-        toggleOnExecute: true,
-        toggleClass: 'tox-tree--directory__label--active'
-      }),
-      ...(visible ? [ Tabstopping.config({}) ] : [])
+      ...(visible ? [ Tabstopping.config({}) ] : []),
+      AddEventsBehaviour.config(directoryLabelEventsId, [
+        AlloyEvents.run<EventArgs<KeyboardEvent>>(NativeEvents.keydown(), (comp, se) => {
+          const isRightArrowKey = se.event.raw.code === 'ArrowRight';
+          const isLeftArrowKey = se.event.raw.code === 'ArrowLeft';
+          if (isRightArrowKey && noChildren ) {
+            se.stop();
+          }
+          if (isRightArrowKey || isLeftArrowKey ) {
+            SelectorFind.ancestor( comp.element, '.tox-tree--directory').each((directoryEle) => {
+              comp.getSystem().getByDom(directoryEle).each((directoryComp) => {
+                if (!Toggling.isOn(directoryComp) && isRightArrowKey || Toggling.isOn(directoryComp) && isLeftArrowKey) {
+                  expandChildren(comp);
+                  se.stop();
+                } else if (isLeftArrowKey && !Toggling.isOn(directoryComp)) {
+                  SelectorFind.ancestor(directoryComp.element, '.tox-tree--directory').each((parentDirElement) => {
+                    SelectorFind.child(parentDirElement, '.tox-tree--directory__label').each((parentDirLabelElement) => {
+                      directoryComp.getSystem().getByDom(parentDirLabelElement).each(Focusing.focus);
+                    });
+                  });
+                  se.stop();
+                }
+              });
+            });
+          }
+        })
+      ])
     ])
-
   });
 };
 
-const renderDirectoryChildren = (children: Dialog.TreeItem[], onLeafAction: OnLeafAction, backstage: UiFactoryBackstage): SimpleSpec => {
-  const computeChildren = (tabstopping: boolean) => children.map((item) => {
-    return item.type === 'leaf' ? renderLeafLabel(item, onLeafAction, tabstopping, backstage) : renderDirectory(item, onLeafAction, tabstopping, backstage);
-  });
+const renderDirectoryChildren = ({
+  children,
+  onLeafAction,
+  visible,
```

Comment:
Is there some way to change add/remove behaviors dynamically without needing to  re-render a new component with the new behaviors every time ?

Reply:
no, I don't think Alloy was designed for the behaviour list to be dynamic - the dynamism happens in the behaviours themselves.